<a href="https://colab.research.google.com/github/Farres26/CIS-3902-AI-MLFARRESAMRHAR/blob/main/Autoviz_(3).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Autoviz
<p><a href="https://github.com/AutoViML/AutoViz">AutoViz documentation in Github</a></p>

Name: Farres Amrhar

Date: 3/23/36

Class: CIS 3902

In [ ]:
!pip install autoviz

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 67.5/67.5 kB 2.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 180.6/180.6 kB 7.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 255.9/255.9 MB 5.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 608.4/608.4 kB 46.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB 71.4 MB/s eta 0:00:00
  Attempting uninstall: xgboost
    Found existing installation: xgboost 3.2.0
    Uninstalling xgboost-3.2.0:
      Successfully uninstalled xgboost-3.2.0


#Seaborn
<p>Seaborn provides several built-in datasets, including "iris", "penguins", "flights", "titanic", "tips", "diamonds", and "exercise", which can be loaded using the seaborn.load_dataset() function. </p>

In [ ]:
import seaborn as sns

In [ ]:
df=sns.load_dataset('tips') #tips is available in seaborn

In [ ]:
df.head()

,total_bill,tip,sex,smoker,day,time,size
0,16.99,1.01,Female,No,Sun,Dinner,2
1,10.34,1.66,Male,No,Sun,Dinner,3
2,21.01,3.50,Male,No,Sun,Dinner,3
3,23.68,3.31,Male,No,Sun,Dinner,2
4,24.59,3.61,Female,No,Sun,Dinner,4


In [ ]:
from autoviz.AutoViz_Class import AutoViz_Class

Imported v0.1.905. Please call AutoViz in this sequence:
    AV = AutoViz_Class()
    %matplotlib inline
    dfte = AV.AutoViz(filename, sep=',', depVar='', dfte=None, header=0, verbose=1, lowess=False,
               chart_format='svg',max_rows_analyzed=150000,max_cols_analyzed=30, save_plot_dir=None)


**Dataset 1: Tips** — Run AutoViz to explore restaurant tip data with `size` as the target variable.

In [ ]:
AV = AutoViz_Class()
dft = AV.AutoViz(
    "", #this is for a filename if you aren't using a dataframe
    sep=",", #separator in the file
    depVar="size", #this is for a target variable name
    dfte=df, #dataframe that has been used for the file that has been input
    header=0, #header row which is typically the first
    verbose=1, #0 minimal charts,  1 is more charts, 2 is for saving locally without display
    lowess=True, #creates regression line with dependent variable
    chart_format="bokeh",
    max_rows_analyzed=150000,
    max_cols_analyzed=30,
    save_plot_dir=None
)

Shape of your Data Set loaded: (244, 7)
#######################################################################################
######################## C L A S S I F Y I N G  V A R I A B L E S  ####################
#######################################################################################
Classifying variables in data set...
    Number of Numeric Columns =  2
    Number of Integer-Categorical Columns =  0
    Number of String-Categorical Columns =  0
    Number of Factor-Categorical Columns =  4
    Number of String-Boolean Columns =  0
    Number of Numeric-Boolean Columns =  0
    Number of Discrete String Columns =  0
    Number of NLP String Columns =  0
    Number of Date Time Columns =  0
    Number of ID Columns =  0
    Number of Columns to Delete =  0
    6 Predictors classified...
        No variables removed since no ID or low-information variables found in data set

################ Multi_Classification problem #####################


Row
    [0] WidgetBox
        [0] Select(name='x', options=['size'], value='size')
        [1] Select(name='y', options=['total_bill', 'tip'], value='total_bill')
    [1] ParamFunction(function, _pane=HoloViews, defer_load=False)

Row
    [0] WidgetBox
        [0] Select(name='X-Axis', options=['total_bill', 'tip'], value='total_bill')
        [1] Select(name='Y-Axis', options=['total_bill', 'tip'], value='tip')
        [2] Select(name='Color', options=['None', 'size'], value='None')
    [1] ParamFunction(function, _pane=HoloViews, defer_load=False)

Column
    [0] Row
        [0] HoloViews(DynamicMap, height=400, sizing_mode='fixed', width=600)
        [1] WidgetBox(align=('end', 'start'))
            [0] Select(options=[np.str_('sex'), ...], value=np.str_('sex'))

Column
    [0] Row
        [0] HoloViews(DynamicMap, height=400, sizing_mode='fixed', width=600)
        [1] WidgetBox(align=('end', 'start'))
            [0] Select(options=[np.str_('total_bill'), ...], value=np.str_('total_bill'))

KDE plot is erroring due to problems with DynamicMaps. Hence it is skipped


HoloViews(Layout)

No date vars could be found in data set


HoloViews(Overlay, height=500, sizing_mode='fixed', width=600)

Row
    [0] WidgetBox
        [0] Select(name='X-Axis', options=['sex', 'smoker', ...], value='sex')
        [1] Select(name='Y-Axis', options=['total_bill', 'tip'], value='total_bill')
    [1] ParamFunction(function, _pane=HoloViews, defer_load=False)

Time to run AutoViz (in seconds) = 9


**Dataset 2: Penguins** — Run AutoViz on penguin measurements with `body_mass_g` as the target variable.

In [ ]:
# Load penguins dataset
df2 = sns.load_dataset('penguins')
df2 = df2.dropna()  # remove missing values
df2.head()

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,Male
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,Female
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,Female
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,Female
5,Adelie,Torgersen,39.3,20.6,190.0,3650.0,Male


In [ ]:
# Run AutoViz on penguins (new instance to avoid conflicts)
AV2 = AutoViz_Class()
dft2 = AV2.AutoViz(
    "",
        sep=",",
            depVar="body_mass_g",  # predict body mass
                dfte=df2,
                    header=0,
                        verbose=1,
                            lowess=False,
                                chart_format="bokeh",
                                    max_rows_analyzed=150000,
                                        max_cols_analyzed=30,
                                            save_plot_dir=None
                                            )

Shape of your Data Set loaded: (333, 7)
#######################################################################################
######################## C L A S S I F Y I N G  V A R I A B L E S  ####################
#######################################################################################
Classifying variables in data set...
    Number of Numeric Columns =  3
    Number of Integer-Categorical Columns =  0
    Number of String-Categorical Columns =  2
    Number of Factor-Categorical Columns =  0
    Number of String-Boolean Columns =  1
    Number of Numeric-Boolean Columns =  0
    Number of Discrete String Columns =  0
    Number of NLP String Columns =  0
    Number of Date Time Columns =  0
    Number of ID Columns =  0
    Number of Columns to Delete =  0
    6 Predictors classified...
        No variables removed since no ID or low-information variables found in data set

################ Regression problem #####################


Column
    [0] Row
        [0] HoloViews(DynamicMap, height=400, sizing_mode='fixed', width=600)
        [1] WidgetBox(align=('end', 'start'))
            [0] Select(options=[np.str_('bill_length_mm')...], value=np.str_('bill_length_mm'))

Row
    [0] WidgetBox
        [0] Select(name='X-Axis', options=['bill_length_mm', ...], value='bill_length_mm')
        [1] Select(name='Y-Axis', options=['bill_length_mm', ...], value='bill_depth_mm')
        [2] Select(name='Color', options=['None', 'body_mass_g'], value='None')
    [1] ParamFunction(function, _pane=HoloViews, defer_load=False)

Column
    [0] Row
        [0] HoloViews(DynamicMap, height=300, sizing_mode='fixed', width=700)
        [1] WidgetBox(align=('end', 'start'))
            [0] Select(options=[np.str_('species'), ...], value=np.str_('species'))

Column
    [0] Row
        [0] HoloViews(DynamicMap, height=400, sizing_mode='fixed', width=600)
        [1] WidgetBox(align=('end', 'start'))
            [0] Select(options=[np.str_('bill_depth_mm'),...], value=np.str_('bill_depth_mm'))

HoloViews(Distribution, height=400, sizing_mode='fixed', width=600)

HoloViews(Violin, height=300, sizing_mode='fixed', width=700)

No date vars could be found in data set


HoloViews(Overlay, height=500, sizing_mode='fixed', width=600)

Row
    [0] WidgetBox
        [0] Select(name='X-Axis', options=['species', 'island', ...], value='species')
        [1] Select(name='Y-Axis', options=['bill_length_mm', ...], value='bill_length_mm')
    [1] ParamFunction(function, _pane=HoloViews, defer_load=False)

Time to run AutoViz (in seconds) = 3


**Summary and Analysis:**

**Dataset 1: Tips**
Restaurant tipping data with 244 records. AutoViz revealed that total_bill and tip are strongly correlated (0.68). Dinner had more tips than lunch, and party size slightly increased the bill.

**Dataset 2: Penguins**
Penguin measurements across 3 species (Adelie, Chinstrap, Gentoo). AutoViz showed clear species clusters based on bill and flipper dimensions. Gentoo penguins have the largest body mass and longest flippers.

#from AutoViz documentation in Github
<p
Arguments for AV.AutoViz() method:<br>

filename: Use an empty string ("") if there's no associated filename and you want to use a dataframe. In that case, using the dfte argument for the dataframe. Otherwise provide a filename and leave dfte argument with an empty string. Only one of them can be used.<br>
sep: File separator (comma, semi-colon, tab, or any column-separating value) if you use a filename above.<br>
depVar: Target variable in your dataset; set it as an empty string if not applicable.<br>
dfte: name of the pandas dataframe for plotting charts; leave it as empty string if using a filename.<br>
header: set the row number of the header row in your file (0 for the first row). Otherwise leave it as 0.<br>
verbose: 0 for minimal info and charts, 1 for more info and charts, or 2 for saving charts locally without display.<br>
lowess: Use regression lines for each pair of continuous variables against the target variable in small datasets; avoid using for large datasets (>100,000 rows).<br>
chart_format: 'svg', 'png', 'jpg', 'bokeh', 'server', or 'html' for displaying or saving charts in various formats, depending on the verbose option.<br>
max_rows_analyzed: Limit the max number of rows to use for visualization when dealing with very large datasets (millions of rows). A statistically valid sample will be used by autoviz. Default is 150000 rows.<br>
max_cols_analyzed: Limit the number of continuous variables to be analyzed. Defaul is 30 columns.<br>
save_plot_dir: Directory for saving plots. Default is None, which saves plots under the current directory in a subfolder named AutoViz_Plots. If the save_plot_dir doesn't exist, it will be created.</p>